src/analysis/calculus/specific_functions.lean

```diff
@@ -240,86 +262,182 @@ smooth_transition.times_cont_diff.times_cont_diff_at
 
 end smooth_transition
 
-variables {E : Type*}
+end real
+
+variable {E : Type*}
+
+/-- `f : times_cont_diff_bump_of_inner c`, where `c` is a point in an inner product space, is a
+bundled smooth function such that
+
+- `f` is equal to `1` in `metric.closed_ball c f.r`;
+- `support f = metric.ball c f.R`;
+- `0 ≤ f x ≤ 1` for all `x`.
 
-/-- Let `x` be a point of a real inner product space; let `0 < r < R` be real numbers.
-Then `smooth_bump_function x r R` is a function `E → ℝ` with the following properties:
+The structure `times_cont_diff_bump_of_inner` contains the data required to construct the function:
+real numbers `r`, `R`, and proofs of `0 < r < R`. The function itself is available through
+`coe_fn`. -/
+structure times_cont_diff_bump_of_inner (c : E) :=
+(r R : ℝ)
+(r_pos : 0 < r)
+(r_lt_R : r < R)
 
-- `smooth_bump_function x r R` is infinitely smooth on `E`;
-- `smooth_bump_function x r R` is equal to `1` on `closed_ball x r`;
-- `0 < smooth_bump_function x r R y < 1` if `r < dist y x < R`;
-- `smooth_bump_function x r R y = 0` if `R ≤ dist y x.
+namespace times_cont_diff_bump_of_inner
 
-We define this function for any `x`, `r`, and `R`  -/
-def smooth_bump_function [inner_product_space ℝ E] (x : E) (r R : ℝ) (y : E) : ℝ :=
-smooth_transition ((R - dist y x) / (R - r))
+lemma R_pos {c : E} (f : times_cont_diff_bump_of_inner c) : 0 < f.R := f.r_pos.trans f.r_lt_R
 
-namespace smooth_bump_function
+instance (c : E) : inhabited (times_cont_diff_bump_of_inner c) := ⟨⟨1, 2, zero_lt_one, one_lt_two⟩⟩
 
-variables [inner_product_space ℝ E] {r R : ℝ} {x y : E}
+variables [inner_product_space ℝ E] {c : E} (f : times_cont_diff_bump_of_inner c) {x : E}
 
-open smooth_transition metric
+instance : has_coe_to_fun (times_cont_diff_bump_of_inner c) :=
+⟨_, λ f x, by exactI real.smooth_transition ((f.R - dist x c) / (f.R - f.r))⟩
```

Comment:
In view of the eager unfolding of coercions by Lean 4, could you instead create a definition `times_cont_diff_bump_of_inner.to_fun := real.smooth_transition ((f.R - dist x c) / (f.R - f.r))`, and register the coercion using `to_fun`? Same thing for all the other partition of unity-like structures in the PR.

Reply:
Done.